.github/workflows/kube-dev.yml

```diff
@@ -0,0 +1,109 @@
+name: Deploy to AWS EKS
+
+on:
+  push:
+    branches: 
+      - tw-738-kubernetes
+
+env:
+  ECR_REPOSITORY: kilt/prototype-chain
+  ECR_IMAGE_TAG: latest-develop
+  SHA_IMAGE_TAG: ${{ github.sha }}
+  KUBECONFIG: '${{ github.workspace }}/.kube/kubeconfig'
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v1
+
+    - name: Configure AWS credentials
+      uses: aws-actions/configure-aws-credentials@v1
+      with:
+        aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        aws-region: eu-central-1
+
+    - name: Login to Amazon ECR
+      id: login-ecr
+      uses: aws-actions/amazon-ecr-login@v1
+
+    - name: Build, tag, and push image to Amazon ECR
+      id: build-image
+      env:
+        ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+        CACHE_IMAGE_BUILDER_TAG: latest-develop-builder
+      run: |
+        docker pull $ECR_REGISTRY/$ECR_REPOSITORY:$CACHE_IMAGE_BUILDER_TAG || true
+        docker build \
+          --target builder \
+          --cache-from $ECR_REGISTRY/$ECR_REPOSITORY:$CACHE_IMAGE_BUILDER_TAG \
+          -t $ECR_REGISTRY/$ECR_REPOSITORY:$CACHE_IMAGE_BUILDER_TAG \
+          .
+        docker pull $ECR_REGISTRY/$ECR_REPOSITORY:$ECR_IMAGE_TAG || true
+        docker build \
+          --cache-from $ECR_REGISTRY/$ECR_REPOSITORY:$CACHE_IMAGE_BUILDER_TAG \
+          --cache-from $ECR_REGISTRY/$ECR_REPOSITORY:$ECR_IMAGE_TAG \
+          -t $ECR_REGISTRY/$ECR_REPOSITORY:$ECR_IMAGE_TAG \
+          -t $ECR_REGISTRY/$ECR_REPOSITORY:$SHA_IMAGE_TAG \
+          .
+        docker push $ECR_REGISTRY/$ECR_REPOSITORY:$CACHE_IMAGE_BUILDER_TAG
+        docker push $ECR_REGISTRY/$ECR_REPOSITORY:$ECR_IMAGE_TAG
+        docker push $ECR_REGISTRY/$ECR_REPOSITORY:$SHA_IMAGE_TAG
+        echo "::set-output name=image::$ECR_REGISTRY/$ECR_REPOSITORY:$ECR_IMAGE_TAG"
+
+    # Here, we're creating the parent directory and writing out our decoded
+    # kubeconfig to the location we stated above.
+    - name: Configure Kubernetes
+      run: |
+        mkdir -p '${{ github.workspace }}/.kube' \
+          && echo '${{ secrets.KUBE_CONFIG}}' | base64 -d > $KUBECONFIG
```

Comment:
oh crazy! 😁 You have the whole config inside a secret!?

Reply:
I was also surprised, that this is how you do it. 😅